ui/components/Filters.js

```diff
@@ -0,0 +1,429 @@
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Divider,
+  Tooltip
+} from "@material-ui/core";
+import { UnControlled as CodeMirror } from "react-codemirror2";
+import DeleteIcon from "@material-ui/icons/Delete";
+import UpdateIcon from "@material-ui/icons/Update";
+import UploadIcon from "@material-ui/icons/Publish";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import MUIDataTable from "mui-datatables";
+import Moment from "react-moment";
+import { withSnackbar } from "notistack";
+import CloseIcon from "@material-ui/icons/Close";
+import MoreHorizIcon from "@material-ui/icons/MoreHoriz";
+import { updateProgress } from "../lib/store";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import dataFetch from "../lib/data-fetch";
+
+const styles = (theme) => ({
+  grid: {
+    padding: theme.spacing(2),
+  },
+  tableHeader: {
+    fontWeight: "bolder",
+    fontSize: 18,
+  },
+});
+
+function CustomToolbar(onClick) {
+  return function Toolbar() {
+    return (
+      <>
+        <label htmlFor="upload-button">
+          <input type="file" accept=".yaml, .yml" hidden onChange={onClick} id="upload-button" name="upload-button" />
+          <Tooltip title="Upload Pattern">
+            <IconButton aria-label="Upload" component="span">
+              <UploadIcon />
+            </IconButton>
+          </Tooltip>
+        </label>
+      </>
+    );
+  };
+}
+
+function YAMLEditor({ pattern, onClose, onSubmit }) {
+  const [yaml, setYaml] = useState("");
+
+  return (
+    <Dialog onClose={onClose} aria-labelledby="pattern-dialog-title" open fullWidth maxWidth="md">
+      <DialogTitle id="pattern-dialog-title">{pattern.name}</DialogTitle>
+      <Divider variant="fullWidth" light />
+      <DialogContent>
+        <CodeMirror
+          value={pattern.pattern_file}
+          options={{
+            theme: "material",
+            lineNumbers: true,
+            lineWrapping: true,
+            gutters: ["CodeMirror-lint-markers"],
+            lint: true,
+            mode: "text/x-yaml",
+          }}
+          onChange={(_, data, val) => setYaml(val)}
+        />
+      </DialogContent>
+      <Divider variant="fullWidth" light />
+      <DialogActions>
+        <Tooltip title="Update Pattern">
+          <IconButton
+            aria-label="Update"
+            color="primary"
+            onClick={() => onSubmit(yaml, pattern.id, pattern.name, "update")}
+          >
+            <UpdateIcon />
+          </IconButton>
+        </Tooltip>
+        <Tooltip title="Delete Pattern">
+          <IconButton
+            aria-label="Delete"
+            color="primary"
+            onClick={() => onSubmit(yaml, pattern.id, pattern.name, "delete")}
+          >
+            <DeleteIcon />
+          </IconButton>
+        </Tooltip>
+      </DialogActions>
+    </Dialog>
+  );
+}
+
+function MesheryPatterns({ updateProgress, enqueueSnackbar, closeSnackbar, user, classes }) {
```

Comment:
Please rename the component name

Reply:
@utkarsh-pro Thanks for reviewing and providing suggestions. I have pushed the changes below (update-filter). Could you please review it again?